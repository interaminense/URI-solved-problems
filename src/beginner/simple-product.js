/**
 * @author Adriano Interaminense <adriano.interaminense@gmail.com>
 * Read two integer values. After this, calculate the product between them and
 * store the result in a variable named PROD.
 * 
 * INPUT
 * 
 * The input file contains 2 integer numbers.
 * @param {number} A
 * @param {number} B
 * 
 * OUTPUT
 * 
 * Print PROD according to the following example, with a blank space before and
 * after the equal signal.
 */
const prod = (A, B) => tmpl(A * B);

/**
 * Template definition
 * @param {string | number} result 
 */
const tmpl = result => `PROD = ${result}`;

export {prod, tmpl};