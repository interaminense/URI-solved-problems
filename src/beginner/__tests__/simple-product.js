import { prod, tmpl } from '../simple-product';

describe('Simple prod', () => {
  let A = 0;
  let B = 0;

  it('The prod 3 * 9 must be 27', () => {
    A = 3;
    B = 9;

    expect(prod(A, B)).toEqual(tmpl(27));
  });

  it('The prod -30 * 10 must be -300', () => {
    A = -30;
    B = 10;

    expect(prod(A, B)).toEqual(tmpl(-300));
  });

  it('The prod 0 * 9 must be 0', () => {
    A = 0;
    B = 9;

    expect(prod(A, B)).toEqual(tmpl(0));
  });
});
