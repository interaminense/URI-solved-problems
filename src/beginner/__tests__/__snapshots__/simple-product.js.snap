// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple prod The prod -30 * 10 must be -300 1`] = `"PROD = -300"`;

exports[`Simple prod The prod 0 * 9 must be 0 1`] = `"PROD = 0"`;

exports[`Simple prod The prod 3 * 9 must be 27 1`] = `"PROD = 27"`;
